palace/linalg/ams.cpp

```diff
@@ -110,7 +111,7 @@ void HypreAmsSolver::ConstructAuxiliaryMatrices(FiniteElementSpace &nd_fespace,
     pi.SetAssemblyLevel(mfem::AssemblyLevel::LEGACY);
     pi.Assemble(skip_zeros_interp);
     pi.Finalize(skip_zeros_interp);
-    ParOperator RAP_Pi(std::unique_ptr<mfem::SparseMatrix>(pi.LoseMat()), h1d_fespace,
+    ParOperator RAP_Pi(std::make_unique<hypre::HypreCSRMatrix>(*pi.LoseMat()), h1d_fespace,
```

Comment:
I would prefer to just get rid of the owning `HypreCSRMatrix` constructor altogether, and just use `pi.SpMat()` here.